### DISCUSSION TITLE
Installing ifuse generates an error

### DISCUSSION THREAD

**Discussion by gooddosh** [Homebrew/discussions#1250] [Everyday usage] [answered]
I'm on Mojave. When I try to install ifuse, I get "Error: ifuse has been disabled because it requires FUSE!". My goal is to mount my iphone on the desktop. Cheers

**Comment by carlocab** [comment, marked as answer]
`ifuse` has been disabled because it has a cask dependency on osxfuse/macfuse. This is no longer supported in homebrew/core.

If you need to use `ifuse`, you can host the formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by gooddosh** [reply]
Thank you. Is there a new recommended method for mounting iOS?

**Reply by carlocab** [reply]
Not one that I'm aware of. You'll have to consult Apple documentation for that.

**Reply by gooddosh** [reply]
LOL, um... Apple definitely does not want us to do this.

**Reply by carlocab** [reply]
Well, I guess that's your answer then :)

**Comment by sgon00** [comment]
I met the same problem when I install encfs. I have written my workaround at this link: https://github.com/vgough/encfs/issues/630#issuecomment-821850213

It's pretty easy, just download the formula .rb file locally, modify one line and then install with the local formula.

Note: you might want to choose osxfuse instead of macfuse. It's your choice.

**Reply by gromgit** [reply]
MacFUSE is definitely the preferred option, unless there's some reason it doesn't install for you, or you cannot accept its license.

**Comment by roz0n** [comment]
I decided to compile `ifuse` from source and it seems to be working well thus far. I wasted a lot of time trying to install it via brew by downloading the formula as described by @sgon00 above but the install failed because it's unable to find the `fuse` dependency despite `osxfuse` being installed without a hitch.

I'm not sure what you require `ifuse` for ultimately but in my case I need it to access Core Data SQLite files inside app sandboxes on a physical device , I figured I'd write up how to go about doing that since anyone trying to do so will likely stumble into this issue thread and be blocked.

Also, this assumes `osxfuse` is installed. Keep in mind that installing it requires a full restart. To ensure it's installed correctly, open the System Preferences panel and you should see two FUSE related items at the bottom. If they aren't there after installing via brew, the FUSE website has a `.pkg` installer that works just as well. Anyway:

1. Download the latest `ifuse` release: `wget https://github.com/libimobiledevice/ifuse/archive/1.1.4.tar.gz`
2. Extract it: `tar -xzvf 1.1.4.tar.gz.1`
3. Navigate into the installer directory and run the autogen script: `cd ifuse-1.1.4 && ./autogen.sh`
4. If you get an error related to `openssl` and `libimobiledevice`, make sure both are installed (e.g. `mdfind openssl`). If they aren't, install them both via brew.
5. If the error persists, paste this into the terminal to set the correct variables:

  ```shell
  export PATH="/usr/local/opt/openssl/bin:$PATH"\nexport LD_LIBRARY_PATH="/usr/local/opt/openssl/lib:$LD_LIBRARY_PATH"\nexport CPATH="/usr/local/opt/openssl/include:$CPATH"\nexport LIBRARY_PATH="/usr/local/opt/openssl/lib:$LIBRARY_PATH"\nexport PKG_CONFIG_PATH="/usr/local/opt/openssl/lib/pkgconfig"
  ```
6. Run the autogen script again, then make and install: `./autogen.sh && make && sudo make install`
7. That should do the trick. Try `ifuse --version`, it should output `ifuse 1.1.4`.

In order to mount a physical device:

1. You'll need the device serial. Install `lsusb` via brew if you don't have it already. The `lsusb` command should output the device serial number if it's plugged in and detected by macOS.
2. Create a mount point of your choice: e.g. `mkdir ~/iPhone`
3. Mount it: `ifuse ~/iPhone -u $SERIAL`

You should see the FUSE device mounted as a volume on your desktop, if not check the Macintosh HD folder in Finder. You can use a tool like `ideviceinstaller` to obtain the app bundle names.

Cheers.

**Reply by carlocab** [reply]
You can also install `ifuse` without needing to compile it (on an Intel Mac, at least) using this tap: https://github.com/gromgit/homebrew-fuse
```
brew install gromgit/fuse/ifuse-mac
```

**Reply by roz0n** [reply]
Thanks a ton, @carlocab. This would've been very useful four hours ago 😅
